File: go-client-sdk/pkg/models/operations/getcompaniescompanyiddatabankaccountsaccountid.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetCompaniesCompanyIDDataBankAccountsAccountIDPathParams struct {
	AccountID string `pathParam:"style=simple,explode=false,name=accountId"`
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
}

type GetCompaniesCompanyIDDataBankAccountsAccountIDQueryParams struct {
	Query *string `queryParam:"style=form,explode=true,name=query"`
}

type GetCompaniesCompanyIDDataBankAccountsAccountIDSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDDataBankAccountsAccountIDRequest struct {
	PathParams  GetCompaniesCompanyIDDataBankAccountsAccountIDPathParams
	QueryParams GetCompaniesCompanyIDDataBankAccountsAccountIDQueryParams
	Security    GetCompaniesCompanyIDDataBankAccountsAccountIDSecurity
}

type GetCompaniesCompanyIDDataBankAccountsAccountIDResponse struct {
	CodatDataContractsDatasetsBankStatementAccount *shared.CodatDataContractsDatasetsBankStatementAccount
	ContentType                                    string
	StatusCode                                     int
}

File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyiddatabankaccountsaccountid.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class GetCompaniesCompanyIdDataBankAccountsAccountIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=accountId" })
  accountId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;
}

export class GetCompaniesCompanyIdDataBankAccountsAccountIdQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=query" })
  query?: string;
}

export class GetCompaniesCompanyIdDataBankAccountsAccountIdSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetCompaniesCompanyIdDataBankAccountsAccountIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdDataBankAccountsAccountIdPathParams;

  @SpeakeasyMetadata()
  queryParams: GetCompaniesCompanyIdDataBankAccountsAccountIdQueryParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdDataBankAccountsAccountIdSecurity;
}

export class GetCompaniesCompanyIdDataBankAccountsAccountIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatDataContractsDatasetsBankStatementAccount?: shared.CodatDataContractsDatasetsBankStatementAccount;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}